[Docs]: fix error in documentation for r/aws_lambda_function (#29110)
hashicorp/terraform-provider-aws: website/docs/r/lambda_function.html.markdown

* `environment` - (Optional) Configuration block. Detailed below.
* `ephemeral_storage` - (Optional) The amount of Ephemeral storage(`/tmp`) to allocate for the Lambda Function in MB. This parameter is used to expand the total amount of Ephemeral storage available, beyond the default amount of `512`MB. Detailed below.
* `file_system_config` - (Optional) Configuration block. Detailed below.

> * `filename` - (Optional) Path to the function's deployment package within the local filesystem. Exactly one of `filename`, `image_uri` and `s3_bucket` must be specified.

Reviewer: I think "or" feels better than "and" here. Oxford comma, love it or hate it 🙂.

```suggestion
* `filename` - (Optional) Path to the function's deployment package within the local filesystem. Exactly one of `filename`, `image_uri`, or `s3_bucket` must be specified.
```

* `handler` - (Optional) Function [entrypoint][3] in your code.
* `image_config` - (Optional) Configuration block. Detailed below.

> * `image_uri` - (Optional) ECR image URI containing the function's deployment package. Exactly one of `filename`, `image_uri` and `s3_bucket` must be specified.

Reviewer: See comment on line 268.

```suggestion
* `image_uri` - (Optional) ECR image URI containing the function's deployment package. Exactly one of `filename`, `image_uri`,  or `s3_bucket` must be specified.
```

* `layers` - (Optional) List of Lambda Layer Version ARNs (maximum of 5) to attach to your Lambda Function. See [Lambda Layers][10]
* `memory_size` - (Optional) Amount of memory in MB your Lambda Function can use at runtime. Defaults to `128`. See [Limits][5]
* `package_type` - (Optional) Lambda deployment package type. Valid values are `Zip` and `Image`. Defaults to `Zip`.
* `publish` - (Optional) Whether to publish creation/change as new Lambda Function Version. Defaults to `false`.
* `reserved_concurrent_executions` - (Optional) Amount of reserved concurrent executions for this lambda function. A value of `0` disables lambda from being triggered and `-1` removes any concurrency limitations. Defaults to Unreserved Concurrency Limits `-1`. See [Managing Concurrency][9]
* `runtime` - (Optional) Identifier of the function's runtime. See [Runtimes][6] for valid values.

> * `s3_bucket` - (Optional) S3 bucket location containing the function's deployment package. This bucket must reside in the same AWS region where you are creating the Lambda function. Exactly one of `filename`, `image_uri` and `s3_bucket` must be specified. Specifying the `s3_key` is manditory when `s3_bucket` is specified.

Reviewer: Again re:line 268. Shortened the last sentence a bit and made it more neutral so that the same wording can be used below.

```suggestion
* `s3_bucket` - (Optional) S3 bucket location containing the function's deployment package. This bucket must reside in the same AWS region where you are creating the Lambda function. Exactly one of `filename`, `image_uri`, or `s3_bucket` must be specified. When `s3_bucket` is set, `s3_key` is required.
```

> * `s3_key` - (Optional) S3 key of an object containing the function's deployment package. Exactly one of `filename`, `image_uri` and `s3_bucket` must be specified. Specifying the `s3_bucket` is manditory when `s3_key` is specified.

Reviewer: Made the last sentence match above, for consistency. I don't think we need to mention the `filename`, `image_uri`, and `s3_bucket` bit here if we're going on relation to the schema. It feels a bit irrelevant when I read it, personally.

Given that you raised this PR, I'm interested in your thoughts on whether we can omit that bit. 

```suggestion
* `s3_key` - (Optional) S3 key of an object containing the function's deployment package. When `s3_bucket` is set, `s3_key` is required.
```
